index.js

```diff
@@ -29,20 +29,40 @@ module.exports = (options) => {
     const socket = {
       connectTimeout: 10000,
       keepAlive: 5000,
+      reconnectStrategy: retries => {
+        logger.info('Reconnect attempt', retries);
+        return 3000;
+      }
     };
     if (config.tls) {
       protocol = 'rediss://';
       socket.tls = true;
       socket.servername = host;
     }
     const url = protocol + host + ':' + config.port + '/' + database;
-    logger.info(`Connecting ${url}`);
     client = redis.createClient({
       url,
       password: config.password,
       socket,
     });
 
+    client.on('connect', () => {
+      logger.info(`Try connecting ${url}`);
+    });
+
+    client.on('ready', () => {
+      logger.info(`Connected ${url}`);
+    })
+
+    // Without handling incoming server errors the process would get finished
+    client.on("error", error => {
+      logger.error("client err", error);
```

Comment:
I think `logger.error` does not support objects as their parameters, which is the reason why we sometimes have some `logger usage error` in some of our APIs. Maybe this would work instead?

```
logger.error(`Client err: ${JSON.stringify(error)}`); 
```